src/vs/server/node/webClientServer.ts

```diff
@@ -183,12 +226,23 @@ export class WebClientServer {
 			accessToken: this._environmentService.args['github-auth'],
 			scopes: [['user:email'], ['repo']]
 		} : undefined;
+		const resourceUrlTemplate = this._productService.extensionsGallery ? URI.parse(this._productService.extensionsGallery.resourceUrlTemplate) : undefined;
 		const data = (await util.promisify(fs.readFile)(filePath)).toString()
 			.replace('{{WORKBENCH_WEB_CONFIGURATION}}', escapeAttribute(JSON.stringify({
 				remoteAuthority,
 				_wrapWebWorkerExtHostInIframe,
 				developmentOptions: { enableSmokeTestDriver: this._environmentService.driverHandle === 'web' ? true : undefined },
 				settingsSyncOptions: !this._environmentService.isBuilt && this._environmentService.args['enable-sync'] ? { enabled: true } : undefined,
+				productConfiguration: <Partial<IProductConfiguration>>{
+					extensionsGallery: resourceUrlTemplate ? {
+						...this._productService.extensionsGallery,
+						'resourceUrlTemplate': resourceUrlTemplate.with({
+							scheme: 'http',
```

Comment:
What if the server runs with `https`, isn't this dynamic?

Reply:
True, since server cannot detect the protocol, this is handled on the client side.